### ISSUE TITLE
[docs] `fp16:min_loss_scale` inconsistency

### ISSUE THREAD

**Issue by stas00** [microsoft/DeepSpeed#1022]
at https://www.deepspeed.ai/docs/config-json/#fp16-training-options 

```
"fp16": {
    [...]
    "min_loss_scale": 1
}
```
and then in the following doc:

```
default: 1000
```

which is 1000x larger than the example.

Which is the recommended value then?

Thank you.

**Comment by kyleliang919** [comment]
I am also very confused about what this is? Is this the scale of absolute value of the loss or the scalar factor that is multiplied on the loss?
the default value of 1 seems to be quite arbitrary and why does it error out when it drops below it? Does it mean the training is overflowing/underflowing? It would be nicer, if there is some insight on this.

**Comment by JamesDConley** [comment]
It seems like the default is 1 and this is the minimum scalar applied to the loss function. E.g. if you set your learning rate to 0.0001 and your min_loss_scale to 1 then your learning rate will be 0.0001 * loss_scale which at lowest will be 1. By default the scale is huge- I've set the starting point to 16 which works out as 2^16 -> 65536.

The learning rate starts at 6.5536 (A very high learning rate) and gets halved each time there is an overflow error until it reaches 1, and if it gets halved again the script crashes. The documentation on loss_scaler.py says it tries to get the highest learning rate it can without causing overflow to avoid underflow.  I'm not an expert on fp16 training specifically- but simply from a training standpoint this seems misguided, since large learning rates can cause instability in training- and IME are also causing overflow problems. 

I wonder if it wouldn't be better to have a a scaler that picks some middle value and raises or lowers the scaling based on whether the loss is nearing underflow or overflow values- trying to stay in the middle rather than riding up against the top edge.

Maybe there is some reason to ride the top edge I am missing here though, if someone from the deepspeed team could give guidance here I'd really appreciate it
